### PR TITLE
#1021 #785 #819 Command migration for HSET, HMSET, HGET, HMGET, HSETNX, HDEL 

### PR DESCRIPTION
**Pull request by anchalsingh25**
PR for #1021, #785, #819
Checklist:
- [x]  Migrated the evalXXX function with the latest definition
- [x]  Update or add unit tests for the new implementation.
- [x]  All unit tests pass successfully.
- [ ]  Ensure all integration tests pass successfully.
- [x] Add Integration tests for the respective commands under the RESP integration tests directory
- [x] Move Integration tests for the respective commands under the RESP integration tests directory from Async directory
- [x] Please validate that the documentation for the respective commands is up to date. If not then consider adding them.

### CODE REVIEW

File: integration_tests/commands/async/hgetall_test.go [hunk 1]
```diff
@@ -7,6 +7,10 @@ import (
 	"gotest.tools/v3/assert"
 )
 
+var ZERO int64 = 0
+var ONE int64 = 1
+var TWO int64 = 2
```
Comment (lucifercr07) [inline integration_tests/commands/async/hgetall_test.go:12]: Where are these vars utilised?

**Review by lucifercr07 — CHANGES_REQUESTED** [review]
@anchalsingh25 if you can follow sample doc [template](https://github.com/DiceDB/dice/blob/c3400e88c8f647392840f1caa83c69c635c1bee7/docs/sample_command_docs.md) and update for these commands accordingly it'd be helpful.

File: internal/worker/cmd_meta.go [hunk 2]
```diff
@@ -107,6 +107,12 @@ const (
 	CmdCMSInitByProb = "CMS.INITBYPROB"
 	CmdCMSMerge      = "CMS.MERGE"
 	CmdCMSIncrBy     = "CMS.INCRBY"
+	CmdHset        = "HSET"
+	CmdHget        = "HGET"
+	CmdHsetnx      = "HSETNX"
+	CmdHdel        = "HDEL"
+	CmdHmset       = "HMSET"
+	CmdHmget       = "HMGET"
```
Comment (apoorvyadav1111) [inline internal/worker/cmd_meta.go:115]:
```suggestion
	CmdHSet        = "HSET"
	CmdHGet        = "HGET"
	CmdHSetNx      = "HSETNX"
	CmdHDel        = "HDEL"
	CmdHMSet       = "HMSET"
	CmdHMGet       = "HMGET"
```

also, please use prettier to keep the space consistent

File: internal/worker/cmd_meta.go [hunk 3]
```diff
@@ -258,6 +264,24 @@ var CommandsMeta = map[string]CmdMeta{
...
+	},
+	CmdHdel: {
+		CmdType: SingleShard,
+	},
+	CmdHmset: {
+		CmdType: SingleShard,
+	},
+	CmdHmget: {
+		CmdType: SingleShard,
+	},
```
Comment (apoorvyadav1111) [inline internal/worker/cmd_meta.go:284]: Please update here as per new vars

**Review by apoorvyadav1111 — CHANGES_REQUESTED** [review]
Hi, thanks for these changes and good to know that you were able to restore the changes. Please check docs for each of these commands and if they follow the structure  as per these samples https://github.com/DiceDB/dice/blob/master/docs/command_docs_template.md
Thanks

File: integration_tests/commands/async/hgetall_test.go [hunk 4]
```diff
@@ -7,6 +7,10 @@ import (
...
+
```
[the remark this answers is not in the document]
Comment (anchalsingh25) [reply integration_tests/commands/async/hgetall_test.go:13]: > Could you please share why this change is needed?

The variables ZERO, ONE, and TWO were defined in async/hset_test.go. When I moved the hset_test.go file from the async directory to the resp directory, these variables became undeclared in multiple files within async. To resolve this, I added these variable to hgetall_test.go.

File: internal/worker/cmd_meta.go (hunk 3 above)
Comment (anchalsingh25) [reply internal/worker/cmd_meta.go:284]: can you elaborate more on this?

File: internal/eval/store_eval.go [hunk 5]
```diff
@@ -2955,3 +2955,255 @@ func evalJSONOBJKEYS(args []string, store *dstore.Store) *EvalResponse {
...
+			Result: nil,
+			Error:  diceerrors.ErrWrongArgumentCount("HSET"),
+		}
+	}
+
+	numKeys, err := insertInHashMap(args, store)
+	if err != nil {
+		return &EvalResponse{
+			Result: nil,
+			Error:  diceerrors.ErrGeneral(err.Error()),
```
[the remark this answers is not in the document]
Comment (anchalsingh25) [reply internal/eval/store_eval.go:3008]: hi @c-harish,  the object type in the hashmap is already being checked in the insertInHashMap function. Could you clarify why we need to check it again.

[the remark this answers is not in the document]
Comment (c-harish) [reply internal/eval/store_eval.go:3008]: Hi, insertinhashmap can return multiple errors (one of it is wrongtype). wrongtype error shouldn't be casted to errgeneral, while other unknown errors should be. no need to check for obj type or cast errors in store_eval, if you can handle it inside the hmap.go
Thanks

File: internal/worker/cmd_meta.go (hunk 3 above)
Comment (apoorvyadav1111) [reply internal/worker/cmd_meta.go:284]: When you rename the vars above, this will be needing an update here as well

**Comment by apoorvyadav1111** [comment]
Hi @anchalsingh25 , looks like some of the unit tests are failing and they could be due to new method of returning error. Please use the following commands to ensure PR gets checks pass.
unit tests:
```
make unittest
```
integration tests:
```
make test
```
linter:
```
golangcli-lint run
```

As you know, we would like to close this soon, please reach out to us on the community if you face any issue.

Thanks

File: integration_tests/commands/resp/hsetnx_test.go [hunk 6]
```diff
@@ -1,9 +1,9 @@
-package async
+package resp
 
 import (
 	"testing"
 
-	"github.com/stretchr/testify/assert"
+	"gotest.tools/v3/assert"
```
Comment (apoorvyadav1111) [inline integration_tests/commands/resp/hsetnx_test.go:6]: why are we going back to assert package here?

**Review by apoorvyadav1111 — CHANGES_REQUESTED** [review]
Thanks for all these changes, please address one minor comment before I merge this. In addition to this, please audit, update or add all the documentations in relevant location (docs folder) using following samples:
`docs/command_docs_template.md`
`docs/sample_command_docs.md`

Please reach out to use on discord for any queries

File: integration_tests/commands/resp/hsetnx_test.go (hunk 6 above)
Comment (anchalsingh25) [reply integration_tests/commands/resp/hsetnx_test.go]: this was unintentional, reverted the changes to keep it consistent.

**Review by apoorvyadav1111 — APPROVED** [review]
Thanks for the changes, Approving

**Pull request merged**